### ISSUE TITLE
🐛 [Bug Report] Creatures cannot regen their mana in combat

### ISSUE THREAD

**Issue by ratkosrb** [cmangos/issues#2021]
## 🐛 Bugreport
Non-pet creatures do not regen mana in combat because they have no intellect.
Bug was introduced here https://github.com/cmangos/mangos-classic/commit/15ea3dc6e1f38e0aa2986dec07bf53a9ce591ca0

sqrt(0) * OCTRegenMPPerSpirit() * ManaIncreaseRate = 0

### Expected behavior
Caster mobs should slowly regen their mana even in combat.

### Version & Environment
Client Version: 1.12
CMaNGOS Repo & Commit Hash:  https://github.com/cmangos/mangos-classic/commit/79001df259f04deca5a663d460ad5b23d6553bfe
Database Repo & Commit Hash: https://github.com/cmangos/classic-db/commit/ee9c247bdc1be2767fcd54da90562fea2d9f73d1
Operating System:  Windows 7 x64

### Steps to reproduce
1. Go to Moonbrook in Westfall
2. Fight a Defias Pillager
3. Wait until he is out of mana
4. Observe that his mana never regens

**Comment by cala** [comment]
Indeed, creatures have no intellect in our stats system. So, we could fix this in https://github.com/cmangos/mangos-classic/commit/15ea3dc6e1f38e0aa2986dec07bf53a9ce591ca0 by:
```diff
-     float intellect = GetStat(STAT_INTELLECT);
+     float intellect = std::max(1, GetStat(STAT_INTELLECT));
```

**Comment by killerwife** [comment]
@cala I like your suggestion very much. Have you tested this solution? Or do we know how much mana creatures regen in combat? Because that might be a formula from whole mana. Unless blizzard uses intelect for mobs, which seems unlikely.

**Comment by ratkosrb** [comment]
While @cala's suggestion does enable some mana regen, it is not the correct amount. With this change a level 15 Defias Pillager will regen 6 mana per tick, while on official classic servers the same mob regens 67 mana.

Classic video:
https://streamable.com/af26o

**Comment by killerwife** [comment]
How much total mana do they have?

**Comment by ratkosrb** [comment]
A level 15 Defias Pillager has 790 mana, as can bee seen from the video. They have the same mana on both cmangos and classic.

**Comment by cala** [comment]
There was an interview of Bill Roper - long ago - where he explained that in early stages of WoW development, they wanted to have a D&D-like stats system for players and NPCs but they later dropped the idea for NPCs in favour of a D2-like stats system for NPCs.

Like @killerwife , I highly doubt NPCs have intelligence because they have no modifiers for it (items nor buffs). The only modifiers they have are for indirect stats like health, mana, AP or damage. This strongly hints that unlike players, they don't have a real stats calculation system (where health is an indirect stats resulting from calculation from direct stats like stamina).

> While @cala's suggestion does enable some mana regen, it is not the correct amount. With this change a level 15 Defias Pillager will regen 6 mana per tick, while on official classic servers the same mob regens 67 mana.

So, if we stick with my suggestion, this would means that in
```c++
    float intellect = std::max(1, GetStat(STAT_INTELLECT));
    addValue = sqrt(intellect) * OCTRegenMPPerSpirit() * ManaIncreaseRate;
```
Either we are returning a wrong value for NPCs in `OCTRegenMPPerSpirit()` or we are missing some multiplier somewhere for creatures.

> A level 15 Defias Pillager has 790 mana, as can bee seen from the video. 

@ratkosrb That's very interesting. Do you use some addon to display the value or is this default client display? I'm asking because in original Classic days, HP and mana values were hidden. Meaning, that many NPCs in our database have incorrect or guessed values. If these values are now available in client, this would be extremely useful!

**Comment by ratkosrb** [comment]
I have the RealMobHealth addon to show the health and mana of the target. Also the server sends the health and mana multipliers in the creature query response packet, and you can use that to calculate it based on cmangos' class level stats table and the amount is exactly the same for this mob. I can submit an update for classic to fix health and mana multipliers for all creatures if you want (currently 3879 mobs with wrong health multiplier, and 405 with wrong mana multiplier in cmangos db).

> Either we are returning a wrong value for NPCs in OCTRegenMPPerSpirit() or we are missing some multiplier somewhere for creatures.

It could be the case that blizzard has a table with default mana regen rates for creatures based on level and unit class, instead of following a stat based formula.

**Comment by cala** [comment]
> I can submit an update for classic to fix health and mana multipliers for all creatures if you want (currently 3879 mobs with wrong health multiplier, and 405 with wrong mana multiplier in cmangos db).

I would really appreciate that, please. That would save us a lot of time. Thanks! :+1: 

> It could be the case that blizzard has a table with default mana regen rates for creatures based on level and unit class, instead of following a stat based formula.

It could be yes. But it could also be a flat % of max mana. That would be my guess and I think that @killerwife think the same, given his previous question. We will probably need more data to make our mind on that topic.

**Comment by ratkosrb** [comment]
I went ahead and tested regen rates on a few more mobs.

Defias Pillager
Total mana: 790
Regen tick: 67
Regen as percent of mana: 8,48% of total mana

Bloodfeather Sorceress
Unit class: 8
Total mana: 350
Regen tick: 39-40
Regen as percent of mana: 8,86%
https://streamable.com/mnp36

Elder Timberling
Unit class: 2
Total mana: 205
Regen tick: 36-37
Regen as percent of mana: 17,80%
https://streamable.com/xt7we

Gordunni Warlock
Unit class: 8
Total mana: 3680
Regen tick: 152
Regen as percent of mana: 4,13%
https://streamable.com/iwh5t

Razorfen Totemic
Unit class: 2
Total mana: 811
Regen tick: 76
Regen as percent of mana:  9,37%
https://streamable.com/6ssje

Razorfen Earthbreaker
Unit class: 2
Total mana: 878
Regen tick: 78-79
Regen as percent of mana: 8,94%
https://streamable.com/z5jgj

Venture Co. Geologist
Unit class: 2
Total mana: 1097
Regen tick: 86
Regen as percent of mana: 7,83%
https://streamable.com/lgksv

Hatecrest Screamer
Unit class: 2
Total mana: 1381
Regen tick: 93-94
Regen as percent of mana: 6.77%
https://streamable.com/cps48
